src/sandbox/algorithms/algorithm.py

```diff
@@ -0,0 +1,14 @@
+from abc import ABC, abstractmethod
+from typing import Any, Callable, TypeVar, Typing, Generic
+import numpy as np
+import gym
+from sandbox.enviroments.multi_armed_bandit.policy import Policy
+
+TValueEstimates = TypeVar('TValueEstimates')
+
+class Algorithm(ABC, Generic[TValueEstimates]):
+
+    @abstractmethod
+    def run(self, n_episodes: int, evniroment: gym.Env, policy: Policy) -> TValueEstimates:
+        ...
```

Comment:
After training algorithm returns a Value estimate which could be either a state-value table, action value table, some neural network trained to approximate action from given state.

Reply:
1. The nice thing about Python docstring is that you can just put it inline in such situations.
Instead of `...` I would put here a docstring with function documentation - it's more practical.
2. evniroment -> environment

Do I understand correctly, this class is responsible only for Policy Evaluation?

Reply:
But i am not sure if is it the best approach, because some algorithms result is learnt policy